landingai/vision_pipeline.py

```diff
@@ -204,6 +210,64 @@ def save_image(self, filename_prefix: str, image_src: str = "") -> "FrameSet":
             c += 1
         return self
 
+    def save_video(
+        self,
+        video_file_path: str,
+        video_fps: Optional[int] = None,
+        video_length_sec: Optional[int] = None,
+        image_src: str = "",
+    ) -> "FrameSet":
+        """Save the FrameSet as an mp4 video file. The following example, shows to use save_video to save a clip from a live RTSP source.
+        ```python
+            video_len_sec=10
+            fps=4
+            img_src = NetworkedCamera(stream_url, fps=fps)
+            frs = FrameSet()
+            for i,frame in enumerate(img_src):
+                if i>=video_len_sec*fps: # Limit capture time
+                    break
+                frs.extend(frame)
+            frs.save_video("sample_images/test.mp4",video_fps=fps)
+        ```
+
+        Parameters
+        ----------
+        video_file_path : str
+            Path and filename with extension of the video file
+        video_fps : Union[int, None], optional
+            Either the frames per second or the video length should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
+        video_length_sec : Union[int, None], optional
+            Either the frames per second or the video length should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
+        image_src : str, optional
+            if empty the source image will be used. Otherwise the image will be selected from `other_images`
+        """
+        if not video_file_path.lower().endswith(".mp4"):
+            raise NotImplementedError("Only .mp4 is supported")
+        total_frames = len(self.frames)
+        if total_frames == 0:
+            return self
+
+        if video_fps is not None and video_length_sec is not None:
+            raise ValueError(
+                "The fps and length arguments cannot be set at the same time"
+            )
+
+        # Try to tune FPS based on parameters or pick a reasonable number
+        if video_length_sec is not None and video_length_sec <= total_frames:
```

Comment:
WDYT if we remove the `video_length_sec <= total_frames` check and make `video_fps` a float? 
(btw, this behavior could cause confusion as the code is not respecting the input value, and it's not documented.)

Reply:
I added a comment to explain the reasoning (which is tricky):

_Try to tune FPS based on parameters or pick a reasonable number. The goal is to produce a video that last a a couple of seconds even when there are few frames. OpenCV will silently fail and not create a file if the resulting fps is less than 1_